ecosystem/platform/server/app/models/user.rb

```diff
@@ -11,18 +11,21 @@ class User < ApplicationRecord
          :omniauthable, omniauth_providers: %i[discord github google],
                         authentication_keys: [:username]
 
+  USERNAME_REGEX = /\A(?!\A[\-_])(?!.*[\-_]{2,})(?!.*[\-_]\Z)[a-zA-Z0-9\-_]+\Z/
+  USERNAME_REGEX_JS = USERNAME_REGEX.inspect[1..-2].gsub('\\A', '^').gsub('\\Z', '$')
```

Comment:
I think `^` and `$` are implicit in html5 `pattern` - not sure if it works if those are specified

Reply:
I can confirm it works as expected!